### DISCUSSION TITLE
DNP DS 620 Prints Full but distorted Images

### DISCUSSION THREAD

**Discussion by Apfelraeuber** [photobooth-app/photobooth-app#335] [Q&A] [answered]
Hi, has anyone yet had the problem that all images by photobooth-app are scaled without keeping the dimensions? I actually tried all different configs of lp and lpr with -o fit-to-page, -o scale=100, custom media format and further more. Nothing seems to have effect. When I realised that the images have 1920x1080 in "full" folder, I also tried to pipe the image through imagemagick by doing something like this: convert input.jpg -resize 1200x1800 JPG:- | lpr -P [Printer_Name] -o fit-to-page

However no effect. Outside of photobooth-app the Prints are perfectly fine. Slowly I am running out of ideas... 😅

**Comment by Apfelraeuber** [comment]
Okay seems like i am getting closer...Resizing does not change the ratio but crop does. Except of loosing parts of the actual image it does now have the right dimensions with: 

convert input.jpg -resize 1800x1200^ -gravity southeast -crop 1800x1200+0+0 +repage JPG:- | lp -o fit-to-page

**Comment by Apfelraeuber** [comment]
Is it possible to reference the original image in the print command? This way I would not loose so much content as by having photobooth-app crop to 16:9 and me cropping back to 4:3 again...

**Comment by skysyk** [comment, marked as answer]
Hmm, you should be able to get the exact output dimensions you need by creating a template image with the dimensions you want and setting the config as needed in mediaprocessing section. I did this to make sure my 4K 16:9 and 24M 3:2 input images were prepared for final printing at 600dpi 3:2 aspect ratio.

If the problem is only with printing and not with processing (the processed image prints the way you like in applications outside of you invoking `lpr` manually), then one thing that may help is to reference the print parameters in any print jobs that turn out properly. I just did this while setting up an RX1HS. Look into collecting CUPS debug log output and then searching for the commandline arguments applied in successful jobs.

**Reply by Apfelraeuber** [reply]
Thanks guys, that was the right hint. I added a watermark weeks ago that had the 1920x1080 aspect ratio to optimize the image for the display. However the watermark file forced a 16:9 ratio which i did not get rid of for printing except for the posted cropping and scaling commands. I now changed the watermark.png to 6:4 to optimize for print and it works perfectly. Only drawback is, that the images in preview do not fully cover the screen. But for the next event, printing is more important. Thanks for your help 💪

**Comment by mgineer85** [comment]
The app does nothing else than executing the command you type in the box.
Only magic is to replace {filename} by the actual file to print.
So once you got it working on the console, it should work in the app also.
About the aspect ratio issues I have no idea unfortunately. You could create your content in the correct ratio already as already suggested or need to crop somehow.
